staging/src/k8s.io/kubectl/pkg/kuberc/kuberc.go

```diff
@@ -0,0 +1,336 @@
+			fmt.Fprintf(errOut, "Warning: Setting alias %q to a built-in command is not supported\n", alias.Name)
+			continue
+		}
+
+		regex := regexp.MustCompile("^[a-zA-Z]+$")
+		if !regex.MatchString(alias.Name) {
+			return args, fmt.Errorf("invalid alias name, can only include alphabetical characters")
+		}
+
+		if _, ok := aliasArgsMap[alias.Name]; ok {
+			fmt.Fprintf(errOut, "alias %s is already set, skipping...\n", alias.Name)
+			continue
+		}
+
+		commands := strings.Split(alias.Command, " ")
+		existingCmd, flags, err := rootCmd.Find(commands)
+		if err != nil {
+			fmt.Fprintf(errOut, "Warning: command %q not found to set alias %q: %v\n", alias.Command, alias.Name, flags)
+			continue
+		}
+
+		newCmd := *existingCmd
+		newCmd.Use = alias.Name
+		aliasCmd := &newCmd
+
+		aliasArgsMap[alias.Name] = struct {
+			args    []string
+			flags   []config.CommandOverrideFlag
+			command *cobra.Command
+		}{
+			args:    alias.Args,
+			flags:   alias.Flags,
+			command: aliasCmd,
+		}
+	}
+
+	// It is verified that all the aliases are valid. So that, now we
+	// can define them in to the root command.
+	for key, val := range aliasArgsMap {
+		p.aliases[key] = struct{}{}
+		rootCmd.AddCommand(val.command)
+	}
+
+	var aliasName string // first "non-flag" arguments
+	for _, arg := range args[1:] {
+		if !strings.HasPrefix(arg, "-") {
+			aliasName = arg
+			break
+		}
+	}
+
+	foundAliasCmd, _, err := rootCmd.Find([]string{aliasName})
+	if err != nil {
+		return args, nil
+	}
+
+	aliasArgs, ok := aliasArgsMap[aliasName]
+	if !ok {
+		return args, nil
+	}
+
+	// This function triggers merging the persistent flags in the parent commands.
+	_ = foundAliasCmd.InheritedFlags()
+
+	for _, fl := range aliasArgs.flags {
+		existingFlag := foundAliasCmd.Flag(fl.Name)
+		if existingFlag == nil {
+			return args, fmt.Errorf("invalid alias flag %s in alias %s", fl.Name, args[0])
+		}
+		if searchInArgs(existingFlag.Name, existingFlag.Shorthand, args) {
+			// Don't modify the value implicitly, if it is passed in args explicitly
+			continue
+		}
+		err = foundAliasCmd.Flags().Set(fl.Name, fl.Default)
+		if err != nil {
+			return args, fmt.Errorf("could not apply value %s to flag %s in alias %s err: %w", fl.Default, fl.Name, args[0], err)
+		}
+	}
+
+	// all args defined in kuberc should be appended to actual args.
+	args = append(args, aliasArgs.args...)
+	// Cobra (command.go#L1078) appends only root command's args into the actual args and ignores the others.
+	// We are appending the additional args defined in kuberc in here and
+	// expect that it will be passed along to the actual command.
+	rootCmd.SetArgs(args[1:])
+	return args, nil
+}
+
+// DefaultGetPreferences returns KubeRCConfiguration.
+// If users sets kuberc file explicitly in --kuberc flag, it has the highest
+// priority. If not specified, it looks for in KUBERC environment variable.
+// If KUBERC is also not set, it falls back to default .kuberc file at the same location
```

Comment:
In alpha, user can only use this feature after exporting `KUBECTL_KUBERC=true`
In beta, user can disable this feature after exporting `KUBECTL_KUBERC=false`
In GA, this will be locked in enabled by default but if there is a necessity, I think we can keep `KUBECTL_KUBERC=false`.

Reply:
@deads2k thoughts on using the envvar to opt out of RC behavior in scripts?

Reply:
I do think we have to give script authors a way to force off RC behavior

Reply:
I agree with that. Even if we promote this functionality to GA, we need to provide a way to easily disable RC behavior. `KUBECTL_KUBERC` is a good candidate for this.

Reply:
I think I agree with envvar (easy to set globally in a script, inert if the script is used with old kubectl versions).

Since we anticipate having an envvar opt-out mechanism around forever, are we happy using the feature gate envvar `KUBECTL_KUBERC`, or do we want to use the `KUBERC` envvar for this (like a special value of `KUBERC` like `KUBERC=off`, similar to `GOPROXY=off`)?

Reply:
In my opinion, we can follow the feature gate convention of kubectl by preserving the `KUBECTL_KUBERC` environment variable in alpha and beta. This can be used to disable the feature anyway. 
In GA, we can remove this environment variable to indicate this feature is promoted to beta and at same time we can introduce `KUBERC=off` environment variable as a disablement method for this GA'ed feature. WDYT?.

Reply:
`KUBERC=off`  would need to be introduced already at beta, so that users forcefully turning this off should be able to use a consistent mechanism. Using `KUBERC` is already supported below, so extending it with `off` capability will be simple addition, and definitely much better than using the feature gate forever.  

@ardaguclu probably worth updating KEP with that comment, as a requirement for beta promotion.

Reply:
Thanks. If we can make it to merge in 1.32, I'll update the KEP.

Reply:
Yeah, that was the original idea, but lemme sync with Arda and try to get it added.